projects/opendr_ws_2/src/opendr_perception/opendr_perception/intent_recognition_node.py

```diff
@@ -0,0 +1,193 @@
+            start_time = perf_counter()
+        if self.last_phrase == data.incremental:
+            # dummy check if new message is the same as previous
+            # to deal with whisper's double messages in short phrases
+            return
+        self.last_phrase = data.incremental
+        # Run intent recognition
+        predictions = self.learner.infer({'text': data.incremental})
+        for pred in predictions:
+            print(LABELS[pred.data])
+        if self.performance_publisher:
+            end_time = perf_counter()
+            fps = 1.0 / (end_time - start_time)  # NOQA
+            fps_msg = Float32()
+            fps_msg.data = fps
+            self.performance_publisher.publish(fps_msg)
+
+        if self.intent_publisher is not None:
+            for prediction in predictions:
+                self.intent_publisher.publish(
+                    self.bridge.to_ros_category(prediction))
+
+
+def main():
+    rclpy.init(args=None)
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--input_transcription_topic",
+        help="Topic name for input transcription",
+        type=str,
+        default="/opendr/transcription")
+    parser.add_argument(
+        "-o",
+        "--output_intent_topic",
+        help="Topic name for output intent",
+        type=lambda value: value if value.lower() != "none" else None,
+        default="/opendr/intent")
+    parser.add_argument(
+        "--performance_topic",
+        help="Topic name for performance messages, disabled (None) by default",
+        type=str,
+        default=None)
+    parser.add_argument(
+        "--device",
+        help="Device to use (cpu, cuda)",
+        type=str,
+        default="cuda",
+        choices=[
+            "cuda",
+            "cpu"])
+    parser.add_argument(
+        "--text_backbone",
+        help="Text backbone that will be used",
+        type=str,
+        default="bert-base-uncased",
+        choices=["bert-base-uncased", "albert-base-v2", "bert-small", "bert-mini", "bert-tiny"])
+
+    parser.add_argument(
+        "--cache_path",
+        help="Path for storing cache",
+        type=str,
+        default="./cache/")
+    args = parser.parse_args()
+
+    try:
+        if args.device == "cuda" and torch.cuda.is_available():
+            device = "cuda"
+        elif args.device == "cuda":
+            print("GPU not found. Using CPU instead.")
+            device = "cpu"
+        else:
+            print("Using CPU.")
+            device = "cpu"
+    except BaseException:
+        print("Using CPU.")
+        device = "cpu"
+
+    intent_node = IntentRecognitionNode(
+        device=device,
+        text_backbone=args.text_backbone,
+        input_transcription_topic=args.input_transcription_topic,
+        output_intent_topic=args.output_intent_topic,
+        performance_topic=args.performance_topic, cache_path=args.cache_path)
+    intent_node.listen()
```

Comment:
Running the node in ROS2 is a bit different.

```suggestion
    intent_node = IntentRecognitionNode(
        device=device,
        text_backbone=args.text_backbone,
        input_transcription_topic=args.input_transcription_topic,
        output_intent_topic=args.output_intent_topic,
        performance_topic=args.performance_topic, cache_path=args.cache_path)

    rclpy.spin(intent_node)

    # Destroy the node explicitly
    # (optional - otherwise it will be done automatically
    # when the garbage collector destroys the node object)
    intent_node.destroy_node()
    rclpy.shutdown()
```